src/gtk/image_gtk.cpp

```diff
@@ -40,7 +42,14 @@ struct BitmapProviderDefault: wxGtkImage::BitmapProvider
 
 double BitmapProviderDefault::GetScale() const
 {
-    return m_win ? m_win->GetDPIScaleFactor() : 1.0;
+    if ( m_win )
+    {
+        return m_win->GetDPIScaleFactor();
+    }
+#if GTK_CHECK_VERSION(3,10,0)
+    return gdk_window_get_scale_factor(wxGetTopLevelGDK());
+#endif
+    return 1.0;
```

Comment:
Aren't we going to get warnings about unreachable code here? We should do this, I think:
```suggestion
#if GTK_CHECK_VERSION(3,10,0)
    return gdk_window_get_scale_factor(wxGetTopLevelGDK());
#else
    return 1.0;
#endif
```